lib/beaker/hypervisor.rb

```diff
@@ -87,6 +88,19 @@ def cleanup
       nil
     end
 
+    #SSH connection method preference. Can be overwritten by hypervisor to change the order
+    def self.connection_preference
+      ['ip', 'vmhostname', 'hostname']
+    end
+
+    def self.set_ssh_connection_perefence(hosts_to_provision, hypervisor)
+      if hypervisor.connection_preference.sort == self.connection_preference.sort
```

Comment:
@rishijavia beaker by default does this twice in an empty run, and even in test suites with hundreds of tests, I don't see it getting called more than 10s of times, so I'm not worried about asymptotic behavior about this.